app/models/katello/subscription.rb

```diff
@@ -24,10 +25,7 @@ def self.using_virt_who
     end
 
     def redhat?
-      # for custom subscriptions, there is no separate marketing and engineering product
-      #   so query our Products table and check there
-      product = Katello::Product.where(:cp_id => self.cp_id, :organization => self.organization).first
-      product.nil? || product.redhat?
```

Comment:
Not sure if the `product.nil?` check was actually meaningful?

Reply:
yeah that check kind of breaks my brain.  What you replaced it with looks much better to me.

Reply:
much cleaner with the right scopes in place!